libsolidity/ast/ASTUtils.cpp

```diff
@@ -18,12 +18,50 @@
 
 #include <libsolidity/ast/AST.h>
 #include <libsolidity/ast/ASTUtils.h>
+#include <libsolidity/ast/ASTVisitor.h>
 
 #include <libsolutil/Algorithms.h>
 
 namespace solidity::frontend
 {
 
+namespace
+{
+
+class ASTNodeLocator: public ASTConstVisitor
+{
+public:
```

Comment:
This will be used in a very very specific way. Does it make sense to change the public interface to only consist of a single static function that takes the offset and returns the ast node pointer?

Reply:
I've deleted this and rewrote `locateInnermostASTNode` to use `SimpleASTVisitor` instead. That should be even better?